plugins/swf-backend/src/service/router.ts

```diff
@@ -162,15 +162,17 @@ function setupInternalRoutes(
       params: { swfId },
     } = req;
     const swfData = req.body;
-    const svcResponse = await executeWithRetry(() =>
-      fetch(`${kogitoBaseUrl}:${kogitoPort}/${swfId}`, {
-        method: 'POST',
-        body: JSON.stringify(swfData),
-        headers: { 'content-type': 'application/json' },
-      }),
+    const svcResponse = await executeWithRetry(
+      () =>
+        fetch(`${kogitoBaseUrl}:${kogitoPort}/${swfId}`, {
+          method: 'POST',
+          body: JSON.stringify(swfData),
+          headers: { 'content-type': 'application/json' },
+        }),
+      [500],
```

Comment:
It seems something somewhere is wrong if `500` is considered a "non-error code".

That said I assume you've investigated whatever problem and this is the necessary fix.

Can you comment why `500` is a "non-error code" for my information?

Reply:
I don't think in this specific case it is because we are converting one API into another API in the proxy, I see the proxy acting as a backend-for-frontend, in this case, the frontend is expecting 201 to redirect to the other page in our case the workflow instances where we can see the started instance in ERROR state..... yeahh confusing anyway without this we get a misleading error page with a stack trace etc, anyway maybe we can think on a best approach but probably requires more changes in scaffolder code that is something we don't want

Reply:
so kogito API returns 500 when the instance was created with an ERROR state, that's the issue, but we have the instanceId in the response, so the instance was created but at some point got an ERROR..